### PR TITLE
[VideoUpload] Add video upload page

### PR DESCRIPTION
**Pull request by edgarlpassos**
## Description

- Adds a page where the user can upload a video by choosing a file and
  a video title
- Adds Protected and Unprotected Routes so that the user is redirected
  from pages when he should not have access to them due to being
  unauthenticated or authenticated

## Testing

- As an unauthenticated user, trying to reach the `/upload` page results in being redirected to the home page
- As an authenticated user, trying to react the `/register` or `/login` pages results in being redirected to the home page
- Manually tried to add a video, the metadata of the video is stored in the DB and the video is stored in S3

### CODE REVIEW

**Comment by GustavJaner** [comment]
I'm trying the code and I get some warnings directly when going to /uploadvideo. Do get have the same?

Maybe the warnings are not important, here they are at least:

Warning: React does not recognize the `bsSize` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bssize` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in div (created by FormGroup)
    in FormGroup (at Login.jsx:54)
    in form (at Login.jsx:53)
    in div (at Login.jsx:52)
    in Login (at UnauthenticatedRoute.jsx:8)
    in Route (at UnauthenticatedRoute.jsx:5)
    in Unknown (at Routes.jsx:15)
    in Switch (at Routes.jsx:12)
    in Unknown (at App.jsx:54)
    in div (at App.jsx:52)
    in App (created by Context.Consumer)
    in withRouter(App) (at src/index.jsx:36)
    in Router (created by BrowserRouter)
    in BrowserRouter (at src/index.jsx:35)

Warning: Failed prop type: The prop `to` is marked as required in `Link`, but its value is `undefined`.
    in Link (created by ForwardRef)
    in ForwardRef (created by NavLink)
    in NavLink (at Navbar.jsx:32)
    in div (created by AbstractNav)
    in AbstractNav (created by Context.Consumer)
    in ContextTransform(AbstractNav) (created by Nav)
    in Nav (at Navbar.jsx:55)
    in nav (created by Navbar)
    in Navbar (created by Uncontrolled(Navbar))
    in Uncontrolled(Navbar) (created by ForwardRef)
    in ForwardRef (created by Bootstrap(undefined))
    in Bootstrap(undefined) (at Navbar.jsx:52)
    in VCNavbar (at App.jsx:53)
    in div (at App.jsx:52)
    in App (created by Context.Consumer)
    in withRouter(App) (at src/index.jsx:36)
    in Router (created by BrowserRouter)
    in BrowserRouter (at src/index.jsx:35)

**Comment by GustavJaner** [comment]
Otherwise the video works perfect. It needs to be added an alert or similar that confirms that the video was uploaded. Since now there is no telling on the client side if something even happened after pressing the "Upload Video" button.
+ also that the form is cleared after pressing the button

**Comment by GustavJaner** [comment]
Also don't know if this is an error, but the error below shows when registrating a new user:
POST https://cognito-identity.eu-central-1.amazonaws.com/ 400

**Comment by GustavJaner** [comment]
Also, now the videos are uploaded with 'private' access level. Thus only readable by the uploader. So we need to change the file access level so any user can see all videos - making the 'home video feed' possible with a collection of videos from all users. 

Now I can only use Storage.get() to get the videos of the currently logged in user.

- Public: Accessible by all users of your app. Files are stored under the public/ path in your S3 bucket.

- Protected: Readable by all users, but writable only by the creating user. Files are stored under protected/{user_identity_id}/ where the user_identity_id corresponds to the unique Amazon Cognito Identity ID for that user.

- Private: Only accessible for the individual user. Files are stored under private/{user_identity_id}/ where the user_identity_id corresponds to the unique Amazon Cognito Identity ID for that user.

**Comment by edgarlpassos** [comment]
> Also, now the videos are uploaded with 'private' access level. Thus only readable by the uploader. So we need to change the file access level so any user can see all videos - making the 'home video feed' possible with a collection of videos from all users.
> 
> Now I can only use Storage.get() to get the videos of the currently logged in user.
> 
> * Public: Accessible by all users of your app. Files are stored under the public/ path in your S3 bucket.
> * Protected: Readable by all users, but writable only by the creating user. Files are stored under protected/{user_identity_id}/ where the user_identity_id corresponds to the unique Amazon Cognito Identity ID for that user.
> * Private: Only accessible for the individual user. Files are stored under private/{user_identity_id}/ where the user_identity_id corresponds to the unique Amazon Cognito Identity ID for that user.

Well spotted. I will change this after we create the video page so I can test it.

**Comment by edgarlpassos** [comment]
> Also don't know if this is an error, but the error below shows when registrating a new user:
> POST https://cognito-identity.eu-central-1.amazonaws.com/ 400

Yeah, I don't know what this error means, or what causes it, but it doesn't seem to affect anything.

**Comment by edgarlpassos** [comment]
> I'm trying the code and I get some warnings directly when going to /uploadvideo. Do get have the same?
> 
> Maybe the warnings are not important, here they are at least:
> 
> Warning: React does not recognize the `bsSize` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bssize` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
> in div (created by FormGroup)
> in FormGroup (at Login.jsx:54)
> in form (at Login.jsx:53)
> in div (at Login.jsx:52)
> in Login (at UnauthenticatedRoute.jsx:8)
> in Route (at UnauthenticatedRoute.jsx:5)
> in Unknown (at Routes.jsx:15)
> in Switch (at Routes.jsx:12)
> in Unknown (at App.jsx:54)
> in div (at App.jsx:52)
> in App (created by Context.Consumer)
> in withRouter(App) (at src/index.jsx:36)
> in Router (created by BrowserRouter)
> in BrowserRouter (at src/index.jsx:35)
> 
> Warning: Failed prop type: The prop `to` is marked as required in `Link`, but its value is `undefined`.
> in Link (created by ForwardRef)
> in ForwardRef (created by NavLink)
> in NavLink (at Navbar.jsx:32)
> in div (created by AbstractNav)
> in AbstractNav (created by Context.Consumer)
> in ContextTransform(AbstractNav) (created by Nav)
> in Nav (at Navbar.jsx:55)
> in nav (created by Navbar)
> in Navbar (created by Uncontrolled(Navbar))
> in Uncontrolled(Navbar) (created by ForwardRef)
> in ForwardRef (created by Bootstrap(undefined))
> in Bootstrap(undefined) (at Navbar.jsx:52)
> in VCNavbar (at App.jsx:53)
> in div (at App.jsx:52)
> in App (created by Context.Consumer)
> in withRouter(App) (at src/index.jsx:36)
> in Router (created by BrowserRouter)
> in BrowserRouter (at src/index.jsx:35)

These are not important. We can fix them later to declutter the console, but they do not affect functionality.

File: src/components/VideoFeed.jsx [hunk 1]
```diff
@@ -28,45 +34,39 @@ class VideoFeed extends Component {
...
     filtered.sort(this.compDate);
-    filtered = filtered.filter((i, index) =>
-      (index < parseInt(this.props.numVideos))
-    );
-    return filtered
+    return filtered.filter((i, index) => (index < numVideos));
   }
 
   filterQuery(videos) {
+    const { owner } = this.props;
```
Comment (GustavJaner) [inline src/components/VideoFeed.jsx:43]: Something changed. Try to load any profile page
eg: http://localhost:3000/profile/36
--> Failed prop type: Invalid prop owner of type string supplied to VideoFeed, expected number.

Now you have to type caste the prop:
const { owner } = parseInt(this.props);

**Review by GustavJaner — COMMENTED** [review]
File permissions working!! Nice

Just one small fix for the Profile page

File: src/components/VideoFeed.jsx (hunk 1 above)
Comment (GustavJaner) [reply src/components/VideoFeed.jsx:43]: Or rather type caste the prop in the Profile.jsx to render the videos correctly in the Profile and to remove the error

//Profile.jsx
  componentDidMount() {
       this.setState({ user: parseInt(this.props.match.params.id) })
  }

Comment (edgarlpassos) [reply src/components/VideoFeed.jsx:43]: I will just pass the prop as a number, since it is a number.

**Pull request merged**